### PR TITLE
Part of #11437: Move GetDateLocal and GetTimeLocal in Platform2.h

### PR DESCRIPTION
**Pull request by frutiemax**

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
I'm not even sure this should be in the platform files as ctime has never changed in the life of the project. but there must have been a 'time' where this made sense.

**Comment by IntelOrca** [comment]
The functions should be changed to return the struct they are writing to rather than using an out parameter.

File: src/openrct2/platform/Platform.Win32.cpp [hunk 1]
```diff
@@ -35,6 +36,7 @@
 #    include "Platform2.h"
 #    include "platform.h"
 
+#    include <ctime>
```
Comment (Gymnasiast) [inline src/openrct2/platform/Platform.Win32.cpp:39]: Is this change necessary? You didn´t change anything else to this file.
